### PR TITLE
Cleanup

### PR DESCRIPTION
**Pull request by dcousens**
Some follow ups to https://github.com/feross/buffer/issues/58

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -105,7 +105,7 @@ function fromString (that, string, encoding) {
   var length = byteLength(string, encoding) | 0
   that = allocate(that, length)
 
-  that.write(string, encoding) | 0
+  that.write(string, encoding)
```
Comment (dcousens) [inline index.js]: Unused result?  Why `| 0`?

File: index.js [hunk 2]
```diff
@@ -114,7 +114,7 @@ function fromObject (that, object) {
 
   if (isArray(object)) return fromArray(that, object)
 
-  if (object == null) {
+  if (object === null || object === undefined) {
```
Comment (dcousens) [inline index.js]: This is [not a performance issue](https://jsperf.com/check-for-null-undefined).
When I made this change, I was under the impression `==` was not allowed in standard.  Apparently it is,  so I'm happy to switch this back.

I personally still prefer the above as a more verbose equivalent, or I prefer the `if (!object)` approach.  Up to you @feross.

File: index.js [hunk 3]
```diff
@@ -855,15 +864,15 @@ Buffer.prototype.writeUIntLE = function writeUIntLE (value, offset, byteLength,
...
-  byteLength = byteLength >>> 0
+  offset = offset | 0
+  byteLength = byteLength | 0
   if (!noAssert) checkInt(this, value, offset, byteLength, Math.pow(2, 8 * byteLength), 0)
 
   var i = byteLength - 1
   var mul = 1
   this[offset + i] = value & 0xFF
   while (--i >= 0 && (mul *= 0x100)) {
-    this[offset + i] = (value / mul) >>> 0 & 0xFF
```
Comment (dcousens) [inline index.js]: `>>> 0` was superfluous here.

File: index.js [hunk 4]
```diff
@@ -230,11 +230,20 @@ Buffer.compare = function compare (a, b) {
 
   var x = a.length
   var y = b.length
-  for (var i = 0, len = Math.min(x, y); i < len && a[i] === b[i]; i++) {}
```
Comment (dcousens) [inline index.js]: No performance issue, but the intention is less clear and IMHO misleading.

File: index.js (hunk 2 above)
Comment (feross) [inline index.js]: I think `== null` is fine.

File: index.js [hunk 5]
```diff
@@ -206,7 +206,7 @@ function checked (length) {
     throw new RangeError('Attempt to allocate Buffer larger than maximum ' +
                          'size: 0x' + kMaxLength.toString(16) + ' bytes')
   }
-  return length >>> 0
+  return length | 0
```
Comment (feross) [inline index.js]: Why? I'd rather keep this the same as the iojs codebase so it's easier to stay in sync.

File: index.js [hunk 6]
```diff
@@ -230,11 +230,20 @@ Buffer.compare = function compare (a, b) {
 
   var x = a.length
   var y = b.length
-  for (var i = 0, len = Math.min(x, y); i < len && a[i] === b[i]; i++) {}
+
+  var i = 0
+  var len = Math.min(x, y)
+  while (i < len) {
+    if (a[i] !== b[i]) break
```
Comment (feross) [inline index.js]: Why not `while (i < len && a[i] === b[i]) {`?

Comment (feross) [inline index.js]: I guess it's a bit clearer the way you wrote it.

File: index.js (hunk 1 above)
Comment (feross) [inline index.js]: Good catch.

File: index.js (hunk 5 above)
Comment (dcousens) [inline index.js]: Consistency?  I'm happy to try and merger this back up stream.

**Comment by feross** [comment]
Tests failed. Rerunning.

File: index.js (hunk 5 above)
Comment (feross) [inline index.js]: If you're interested, go for it.

**Comment by feross** [comment]
Tests passed. I rebased to remove your `== null` changes and I'll publish a new patch version as soon as my local tests pass.

**Comment by dcousens** [comment]
Sounds good :)

Cheers @feross

**Comment by feross** [comment]
Published as 3.2.1.

**Comment by dcousens** [comment]
By the way, this PR's branch was in this repository, feel free to collaborate/rebase the PR's branch in future rather than rebasing externally.  

Something satisfying about that purple "success" after a PR merge haha.  Also more visible from a notifications POV.

**Comment by feross** [comment]
Will do next time! Thanks :-)

**Pull request closed without merging**